framework/include/userobject/DomainUserObject.h

```diff
@@ -0,0 +1,106 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "UserObject.h"
+#include "BlockRestrictable.h"
+#include "TwoMaterialPropertyInterface.h"
+#include "NeighborCoupleable.h"
+#include "MooseVariableDependencyInterface.h"
+#include "TransientInterface.h"
+#include "RandomInterface.h"
+#include "ElementIDInterface.h"
+#include "MooseError.h"
+
+namespace libMesh
+{
+class Elem;
+class QBase;
+}
+
+class DomainUserObject : public UserObject,
+                         public BlockRestrictable,
+                         public TwoMaterialPropertyInterface,
+                         public NeighborCoupleable,
+                         public MooseVariableDependencyInterface,
+                         public TransientInterface,
+                         public RandomInterface,
+                         public ElementIDInterface
+{
+public:
+  static InputParameters validParams();
+
+  DomainUserObject(const InputParameters & parameters);
+
+  void execute() override;
+
+  /**
+   * execute method that is called during ComputeUserObjects::onElement
+   */
+  virtual void executeOnElement() {}
+
+  /**
+   * execute method that is called during ComputeUserObjects::onBoundary
+   */
+  virtual void executeOnBoundary() {}
+
+  /**
+   * execute method that is called during ComputeUserObjects::onInternalSide
+   */
+  virtual void executeOnInternalSide() {}
```

Comment:
Do we need `executeOnInterface`?

Reply:
Actually I was reviewing as a whole and now noticed that you had that method before and removed it later.